SaltSquareIO/sports-app-react-fe
# User profile UI on the homepage

[CLICKUP_TICKET_ID](https://app.clickup.com/t/86c0g7mvu)

## Summary of change

This PR implements the user profile UI on the home page. Key changes include:

1. Added an app bar to both the home and profile page, providing seamless navigation between both pages. This bar improves accessibility and maintains a consistent design across pages.
2. The profile page displays key user information fetched from the `/user/me` endpoint. 
3. Introduced a dedicated error page. If data retrieval from `/user/me` fails, the error page is displayed with guidance to return to the home page. The page is only accessible when triggered by an error and cannot be accessed directly via URL.

## QA Notes

To test this PR, follow these instructions: 

1. Verify that the "User Profile" link in the app bar navigates to the profile page.
2. Confirm that the user data from `/user/me` displays correctly.
3. Test the "Home" link in the app bar to ensure it returns to the home page.
4. Simulate a failed data request to verify that the error page is displayed. 
5. Make sure the "Back to home" button on the error page navigates to the home page. 

## Screenshots

### User Profile 

<img width="1904" alt="Screenshot 2024-10-29 at 10 57 20" src="https://github.com/user-attachments/assets/a9647b4d-3cb5-47c1-9217-81422652ae5c">

### Error while loading user data

<img width="1904" alt="Screenshot 2024-10-29 at 10 57 45" src="https://github.com/user-attachments/assets/36ce4b5c-d7c8-4952-99ac-ba7ed36610b3">
